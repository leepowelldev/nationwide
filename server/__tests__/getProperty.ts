import { gql } from 'apollo-server';
import { mockCollection } from './__utils__/mockCollections.js';
import setup from './__utils__/setup.js';

const query = gql`
  query ($id: ID!) {
    getProperty(id: $id) {
      id
      address
      type
      bedrooms
    }
  }
`;

describe('Query: getProperty', () => {
  test('should return property', async () => {
    const { server } = setup(mockCollection);

    const res = await server.executeOperation({
      query,
      variables: { id: '1' },
    });

    expect(res).toMatchSnapshot();
  });

  test('should error when property does not exist', async () => {
    const { server } = setup(mockCollection);

    const res = await server.executeOperation({
      query,
      variables: { id: '2' },
    });

    expect(JSON.stringify(res.errors?.[0])).toMatchSnapshot();
    expect(res).toMatchSnapshot();
  });
});
