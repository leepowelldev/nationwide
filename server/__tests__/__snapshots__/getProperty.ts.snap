// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query: getProperty should error when property does not exist 1`] = `"{\\"message\\":\\"property with id 2 does not exist\\",\\"locations\\":[{\\"line\\":2,\\"column\\":3}],\\"path\\":[\\"getProperty\\"],\\"extensions\\":{\\"code\\":\\"NOT_FOUND\\"}}"`;

exports[`Query: getProperty should error when property does not exist 2`] = `
Object {
  "data": Object {
    "getProperty": null,
  },
  "errors": Array [
    [GraphQLError: property with id 2 does not exist],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Query: getProperty should return property 1`] = `
Object {
  "data": Object {
    "getProperty": Object {
      "address": "A test address",
      "bedrooms": 4,
      "id": "1",
      "type": "HOUSE",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
